qualtran/simulation/quimb_sim.py

```diff
@@ -104,8 +104,10 @@ def _assign_outgoing(cxn: Connection) -> Soquet:
             return cxn.right
         return cxn.left
 
+    visited_bloqs: Set[BloqInstance] = set()
     for binst, incoming, outgoing in cbloq.iter_bloqnections():
         bloq = binst.bloq
+        visited_bloqs.add(binst)
```

Comment:
I think the correct fix would be to modify `iter_bloqnections` to yield all the bloqs with empty `incoming` and `outgoing`.

Reply:
done